RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -1,483 +1,505 @@
+                         &barrelRecHits, &endcapRecHits, hcalTowersHandle,
+                         vertexCollection, outputPhotonCollection, iSC,
+                         sevLv.product());
 
   // put the product in the event
-  edm::LogInfo("PhotonProducer") << " Put in the event " << iSC << " Photon Candidates \n";
-  outputPhotonCollection_p->assign(outputPhotonCollection.begin(),outputPhotonCollection.end());
+  edm::LogInfo("PhotonProducer")
+      << " Put in the event " << iSC << " Photon Candidates \n";
+  outputPhotonCollection_p->assign(outputPhotonCollection.begin(),
+                                   outputPhotonCollection.end());
   theEvent.put(std::move(outputPhotonCollection_p), PhotonCollection_);
-
 }
 
-void PhotonProducer::fillPhotonCollection(edm::Event& evt,
-					  edm::EventSetup const & es,
-					  const edm::Handle<reco::PhotonCoreCollection> & photonCoreHandle,
-					  const CaloTopology* topology,
-					  const EcalRecHitCollection* ecalBarrelHits,
-					  const EcalRecHitCollection* ecalEndcapHits,
-					  const edm::Handle<CaloTowerCollection> & hcalTowersHandle,
-					  // math::XYZPoint & vtx,
-                                          reco::VertexCollection & vertexCollection,
-					  reco::PhotonCollection & outputPhotonCollection, int& iSC,
-					  const EcalSeverityLevelAlgo * sevLv) {
-
+void PhotonProducer::fillPhotonCollection(
+    edm::Event& evt, edm::EventSetup const& es,
+    const edm::Handle<reco::PhotonCoreCollection>& photonCoreHandle,
+    const CaloTopology* topology, const EcalRecHitCollection* ecalBarrelHits,
+    const EcalRecHitCollection* ecalEndcapHits,
+    const edm::Handle<CaloTowerCollection>& hcalTowersHandle,
+    reco::VertexCollection& vertexCollection,
+    reco::PhotonCollection& outputPhotonCollection, int& iSC,
+    const EcalSeverityLevelAlgo* sevLv) {
   const CaloGeometry* geometry = theCaloGeom_.product();
-  const CaloSubdetectorGeometry* subDetGeometry =0 ;
-  const CaloSubdetectorGeometry* geometryES = theCaloGeom_->getSubdetectorGeometry(DetId::Ecal, EcalPreshower);
-  const EcalRecHitCollection* hits = 0 ;
+  const CaloSubdetectorGeometry* subDetGeometry = 0;
+  const CaloSubdetectorGeometry* geometryES =
+      theCaloGeom_->getSubdetectorGeometry(DetId::Ecal, EcalPreshower);
+  const EcalRecHitCollection* hits = 0;
   std::vector<double> preselCutValues;
-  float minR9=0;
-
+  float minR9 = 0;
 
   std::vector<int> flags_, severitiesexcl_;
 
-  for(unsigned int lSC=0; lSC < photonCoreHandle->size(); lSC++) {
-
+  for (unsigned int lSC = 0; lSC < photonCoreHandle->size(); lSC++) {
     reco::PhotonCoreRef coreRef(reco::PhotonCoreRef(photonCoreHandle, lSC));
-    reco::SuperClusterRef scRef=coreRef->superCluster();
-    //    const reco::SuperCluster* pClus=&(*scRef);
+    reco::SuperClusterRef scRef = coreRef->superCluster();
     iSC++;
 
+    bool isHGCal = (scRef->seed()->seed().det() == DetId::Forward) ||
```

Comment:
I'm not sure: are you suggesting to move to the new C++ producer GEDPhotonProduce instead of PhotonProducer?
Maybe, if that's the case, that could be done at a later stage? 
That's definitely more in the hands of EGamma POG, IMHO.